sklearn/kernel_approximation.py

```diff
@@ -334,7 +334,7 @@ def fit(self, X, y=None):
         self : object
             Returns the transformer.
         """
-        X = check_array(X, accept_sparse='csr')
+        check_array(X, accept_sparse='csr')
```

Comment:
I was like "this definitely doesn't look right" but it actually surprisingly is. With estimator tags we could entirely remove this line.

Reply:
Not familiar with estimator tags, but are the tags in place so this line can be removed or should this be kept for now to still validate `X`?

Reply:
should be kept for now, sorry for being cryptic.